x-pack/platform/plugins/shared/actions/server/lib/gen_ai_token_tracking.ts

```diff
@@ -284,13 +305,31 @@ export const getGenAiTokenTracking = async ({
         total_tokens: usage.totalTokens,
         prompt_tokens: usage.inputTokens,
         completion_tokens: usage.outputTokens,
+        telemetry_metadata: telemetryMetadata,
       };
     } else {
       logger.error('Response from Bedrock converse API did not contain usage object');
       return null;
     }
   }
 
+  if (actionTypeId === '.bedrock' && validatedParams.subAction === 'invokeAIRaw') {
+    const results = result.data as unknown as {
+      content: Array<{ type: string; text: string }>;
+      usage?: { input_tokens: number; output_tokens: number };
+    };
+    if (results?.usage) {
+      return {
+        total_tokens: results.usage.input_tokens + results.usage.output_tokens,
```

Comment:
There's no typechecking on `usage` or `usage.input_tokens` and `usage.output_tokens`, so the `*_tokens` values could (in theory) be literally anything.  Eg, `total_tokens` could end up being `NaN`.  I'd suggest a deconstructed assignment of the `*_tokens` properties, and a number check on them ...

Reply:
Added in https://github.com/elastic/kibana/pull/208180/commits/978ae5b6a9042a97c0c2e9afa512dca175e54051.

Please note that no other code in that specific location is doing something similar, which is the reason I tried to rather mimick the rest of the implementation. I do agree it should be doing a better job on typechecking and safe access, but it will stand out a bit in terms of the rest of the usage count.